### ISSUE TITLE
Dataset list: schema not allowed to filter

### ISSUE THREAD

**Issue by xingyuli** [apache/superset#22203]
On the `Datasets` page, I cannot search by utilizing the SCHEMA filter.

#### How to reproduce the bug

1. Go to 'Data -> Datasets'
2. Choose from the 'SCHEMA' drop down list.
3. See error

### Expected results

Should be work.

### Actual results

It just says 'HTTP 400 bad requests' ...

#### Screenshots

![image](https://user-images.githubusercontent.com/3337431/203502261-fbc57462-8497-4065-8189-0387e3cab7fd.png)

Superset log:

```
[23/Nov/2022:16:34:12 +0800] "GET /api/v1/dataset/?q=(filters:!((col:schema,opr:eq,value:information_schema)),order_column:changed_on_delta_humanized,order_direction:desc,page:0,page_size:25) HTTP/1.1" 400 58 "http://superset.k8s-dev/tablemodelview/list/?filters=(schema:(label:information_schema,value:information_schema))&pageIndex=0&sortColumn=changed_on_delta_humanized&sortOrder=desc" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36"
```

### Environment

(please complete the following information):

- superset version: `2.0.0`

**Comment by stephanclaus** [comment]
we have the same problem as well, would be great to get a fix for this one.

**Comment by eschutho** [comment]
I confirmed that this issue is present on 2.0 but fixed in master, so I'm going to close this as resolved, and we'll get the fix out in 2.0.2 or 2.1.0. Thanks for flagging it!
